x-pack/test/functional/page_objects/navigational_search.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { WebElementWrapper } from '../../../../test/functional/services/lib/web_element_wrapper';
+import { FtrProviderContext } from '../ftr_provider_context';
+
+interface SearchResult {
+  label: string;
+}
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export function NavigationalSearchProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+
+  class NavigationalSearch {
+    async focus() {
+      const field = await testSubjects.find('nav-search-input');
+      await field.click();
+    }
+
+    async blur() {
+      await testSubjects.click('helpMenuButton');
+      await testSubjects.click('helpMenuButton');
+      await find.waitForDeletedByCssSelector('.navSearch__panel');
+    }
+
+    async searchFor(
+      term: string,
+      { clear = true, wait = true }: { clear?: boolean; wait?: boolean } = {}
+    ) {
+      if (clear) {
+        await this.clearField();
+      }
+      const field = await testSubjects.find('nav-search-input');
+      await field.type(term);
+      if (wait) {
+        await this.waitForResultsLoaded();
+      }
+    }
+
+    async clearField() {
+      const field = await testSubjects.find('nav-search-input');
+      await field.clearValueWithKeyboard();
+    }
+
+    async isPopoverDisplayed() {
+      return await find.existsByCssSelector('.navSearch__panel');
+    }
+
+    async clickOnOption(index: number) {
+      const options = await testSubjects.findAll('nav-search-option');
+      await options[index].click();
+    }
+
+    async waitForResultsLoaded(waitUntil: number = 3000) {
+      await testSubjects.exists('nav-search-option');
+      // results are emitted in multiple batches. Each individual batch causes a re-render of
+      // the component, causing the current elements to become stale. We can't perform DOM access
+      // without heavy flakiness in this situation.
+      // there is NO ui indication of any kind to detect when all the emissions are done,
+      // so we are forced to fallback to awaiting a given amount of time once the first options are displayed.
+      await delay(waitUntil);
```

Comment:
Self explanatory comment. I can't rely on the UI at all to known when all the results are loaded: there are no indication that the searchbar did not received all the results yet, so I'm forced to use a good old delay here.

I will run that against the flaky runner to be sure than the 3000ms is enough.

Reply:
Should we add a `data-UI-render-complete` prop to the popover or something so that we can rely on that in the test instead of a delay?

Reply:
I wouldn't be able to use that from a FTR test. it's a user test, I would need something a human could see from the UI.

Reply:
Wait, isn't the Elastic logo that indicator? It stays spinning (as a loading indicator) while search is happening.

Reply:
Nope. There was a PR that added a delay before displaying that indicator, so now we can't use it for that anymore. (#78879)